source/importers/SynchronizeColorsTokens.ts

```diff
@@ -0,0 +1,218 @@
+			const patternColorsToProcess = /RAW\/|theme/i;
+			const patternColorsToIgnore = /deprecated|Pattern/i;
+			const patternTheme = /theme/i;
+			if (obj.name.match(patternColorsToProcess) && obj.name.match(patternColorsToIgnore) === null) {
+				const nameSplit = obj.name.split('/');
+				let colorName = (nameSplit[2] === undefined) ? nameSplit[1] : nameSplit[2];
+				console.log(colorName);
+				obj.values.hex = obj.values.hex.replace(/FFFFFF/gi, function(){
+					return 'fff';
+				});
+
+				const colorFinalName = obj.name.match(patternTheme) ? '--'  + colorName : '--raw-' + colorName;
+				console.log(colorFinalName);
+
+				if (hexToCssVariable[obj.values.hex] === undefined) {
+					hexToCssVariable[obj.values.hex] = colorFinalName ;
+				}
+
+				result.push(colorFinalName + ': ' + obj.values.hex + ';');
+				result.push(colorFinalName + '-rgb' + ': ' + hexToRgb(obj.values.hex) + ';');
+
+				/** JSON object structure */
+				colorName = colorName.replace(/--/i, '').replace(/raw-/i, '');
+				const colorNameSplitted = colorName.split('-');
+				const category = (colorNameSplitted[1] === undefined) ? 'default' : colorNameSplitted[0];
+				const resultJsonObject: IResultJsonObject = {
+					id: binValues.destination + '_' + colorName,
+					label: colorName,
+					value: obj.values.hex,
+					weight: parseInt(colorNameSplitted[1])
+				};
+
+				if (temporaryColorsJson[category] === undefined) {
+					temporaryColorsJson[category] = [];
+				}
+				temporaryColorsJson[category].push(resultJsonObject);
+				temporaryColorsJson[category].sort((a,b) => a.weight - b.weight);
+			}
+		});
+
+		result.push('}\n');
+
+		Object.entries<Array<IResultJsonObject>>(temporaryColorsJson).forEach(([category, colors]) => {
+			Object.entries(colors).forEach(([, color]) => {
+				const resultFileJsonObject: IResultFileJsonObject = {
+					id: color.id,
+					label: color.label,
+					value: color.value
+				}
+
+				if (resultColorsJson[category] === undefined) {
+					resultColorsJson[category] = [];
+				}
+				resultColorsJson[category].push(resultFileJsonObject);
+			});
+		});
+
+		arrayToFile(tokensFilesConfig.destinationPath + binValues.destination, result);
+		jsonToFile(tokensFilesConfig.destinationPath + binValues.destinationJson, resultColorsJson);
+
+	} catch (err) {
+		console.error(err)
+	}
+
+	return hexToCssVariable;
+};
+
+const ImportSingleTokenFile = (
+	binValues : configFileObject,
+	jsonColors: Array<any>,
+	hexToCssVariable: Object,
+) => {
+	try {
+		let result: Array<string> = [];
+		let resultJson = {};
+
+		jsonColors.forEach(obj => {
+			const patternColorsToIgnore = /deprecated|Pattern|RAW|theme/i;
+			if (obj.name.match(patternColorsToIgnore) === null) {
+				let tokenName = obj.name;
+				tokenName = tokenName.replace(/\//i, '-');
+
+				if (obj.values.hex .split('').every(char => char === obj.values.hex [0])){
+					obj.values.hex  = `${obj.values.hex [0]}${obj.values.hex [1]}${obj.values.hex [2]}`
```

Comment:
these random spaces looks strange, did you consider using `prettier` on save https://bethink.atlassian.net/wiki/spaces/LMS/pages/137265159/Korzystanie+z+formatera+na+froncie ?

Reply:
Updated.